src/sinks/elasticsearch/config.rs

```diff
@@ -38,41 +38,89 @@ use lookup::path;
 /// The field name for the timestamp required by data stream mode
 pub const DATA_STREAM_TIMESTAMP_KEY: &str = "@timestamp";
 
-#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+/// Configuration for the `elasticsearch` sink.
+#[configurable_component(sink)]
+#[derive(Clone, Debug, Default)]
 #[serde(deny_unknown_fields)]
 pub struct ElasticsearchConfig {
+    /// The Elasticsearch endpoint to send logs to.
+    ///
+    /// This should be the full URL as shown in the example.
```

Comment:
🤔 unrelated to your PR, but I wonder why this isn't a more specific type than String.

Reply:
It should be `UriSerde` or similar, but I decided to leave it as-is for the moment.

Reply:
@jszwedko could we call this out as something to check as part of the integrations audit?